src/ace_time/hw/StmRtc.cpp

```diff
@@ -17,13 +17,18 @@ using ace_common::decToBcd;
 namespace ace_time {
 namespace hw {
 
-StmRtc::StmRtc() {
+StmRtc::StmRtc() {}
 
+bool StmRtc::begin(const sourceClock_t clockSource, const hourFormat_t hourFormat) {
+  
   mRtc = &STM32RTC::getInstance();
   if (mRtc) {
-    mRtc->setClockSource(STM32RTC::LSI_CLOCK);
-    mRtc->begin(HOUR_FORMAT_24);
+  // Select RTC clock source: LSI_CLOCK, LSE_CLOCK or HSE_CLOCK.
+    mRtc->setClockSource((STM32RTC::Source_Clock) clockSource);
```

Comment:
If we used `STM32RTC::Source_Clock` in the signature for `begin()` (as I suggest below), we wouldn't need this cast.